File: file-handling/handling-text-files/visual-basic/read_text_file_line_by_line.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        ReadLineByLine()
    End Sub

    Sub ReadLineByLine()
         Using reader As System.IO.StreamReader = New System.IO.StreamReader("bridge.txt")
             Dim line As String
             line = reader.ReadLine()
             While line IsNot Nothing ' While not end of file
                Console.WriteLine(line) 
                line = reader.ReadLine() ' Read next ine of file
             End While
         End Using
     End Sub


End Module

File: file-handling/handling-text-files/java/read_text_file_line_by_line.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science

Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

class HandlingTextFiles
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args) {
        readLineByLine();
    }


    // Read and display each record from a file line by line
    public static void readLineByLine() {
        // Uses try-with-resources to automatically close the readers after execution
        try (FileReader fr = new FileReader("bridge.txt");
             BufferedReader reader = new BufferedReader(fr)) {

            String line = reader.readLine();

            while (line != null) {
                System.out.println(line);
                line = reader.readLine(); // Read the next line
            }
        } // The readers are now closed
        catch (IOException e) {
            e.printStackTrace();
        }
    }


}